pkg/nsx/services/vpc/vpc.go

```diff
@@ -521,53 +536,64 @@ func (s *VPCService) DeleteLBPool(path string) error {
 }
 
 func (s *VPCService) IsSharedVPCNamespaceByNS(ctx context.Context, ns string) (bool, error) {
-	sharedNS, err := s.getSharedVPCNamespaceFromNS(ctx, ns)
+	_, sharedNamespaceObj, err := s.resolveSharedVPCNamespace(ctx, ns)
 	if err != nil {
 		if apirrors.IsNotFound(err) {
 			return false, nil
 		}
 		return false, err
 	}
-	if sharedNS == "" {
+	if sharedNamespaceObj == nil {
 		return false, nil
 	}
-	if sharedNS != ns {
+	if sharedNamespaceObj.Name != ns {
 		return true, nil
 	}
 	return false, err
 }
 
-func getNamespace(c client.Client, ctx context.Context, ns string) (*v1.Namespace, error) {
+func (s *VPCService) getNamespace(ctx context.Context, name string) (*v1.Namespace, error) {
 	obj := &v1.Namespace{}
-	if err := c.Get(ctx, types.NamespacedName{Name: ns}, obj); err != nil {
-		log.Error(err, "Failed to fetch Namespace", "Namespace", ns)
+	if err := s.Client.Get(ctx, types.NamespacedName{Name: name}, obj); err != nil {
+		log.Error(err, "Failed to fetch Namespace", "Namespace", name)
 		return nil, err
 	}
 	return obj, nil
 }
 
-func (s *VPCService) getSharedVPCNamespaceFromNS(ctx context.Context, ns string) (string, error) {
-	obj, err := getNamespace(s.Client, ctx, ns)
+// resolveSharedVPCNamespace will resolve the Namespace relationship based on VPC sharing,
```

Comment:
I didn't get your point why this function must return both the original k8s NS object and the shared K8s NS object (if exists). If the purpose of this change is to return the K8s NS object (rather than the NS name) which is used to query the NSX VPC, I think you  can just add a 2nd K8s get operation to read the shared NS object based on the old implementation?

Reply:
If we use the old implementation method, we need to call the 'Get' K8s API twice to get the Na mespace(UID).

Reply:
I mean the consumer actually only cares which K8s Namespace object can be used, but doesn't care it is a shared Namespace or its own Namespace, Why we must return two objects (plus error) in this function?